client/jetpack-cloud/components/sidebar/site-selector.tsx

```diff
@@ -0,0 +1,62 @@
+import { useRef } from 'react';
+import BaseSiteSelector from 'calypso/components/site-selector';
+import { useDispatch, useSelector } from 'calypso/state';
+import { hasJetpackPartnerAccess } from 'calypso/state/partner-portal/partner/selectors';
+import { setLayoutFocus } from 'calypso/state/ui/layout-focus/actions';
+import { getCurrentLayoutFocus } from 'calypso/state/ui/layout-focus/selectors';
+import useOutsideClickCallback from './use-outside-click-callback';
+
+/* NOTE: Code for this component was borrowed from calypso/my-sites/picker,
+ * with some slight modifications because we can safely assume we're using
+ * Jetpack Cloud.
+ */
+
+const SITES_FOCUS = 'sites';
+
+const scrollToTop = () => {
+	document.getElementById( 'secondary' ).scrollTop = 0;
+	window.scrollTo( 0, 0 );
+};
+
+const useHideSiteSelectorOnFocusOut = () => {
+	const dispatch = useDispatch();
+	const isVisible = useSelector( ( state ) => getCurrentLayoutFocus( state ) === SITES_FOCUS );
+	const siteSelectorRef = useRef( null );
+
+	useOutsideClickCallback( siteSelectorRef, () => {
+		if ( ! isVisible ) {
+			return;
+		}
+
+		dispatch( setLayoutFocus( 'sidebar' ) );
+		scrollToTop();
+	} );
+
+	return siteSelectorRef;
+};
+
+const SiteSelector = () => {
+	const isVisible = useSelector( ( state ) => getCurrentLayoutFocus( state ) === SITES_FOCUS );
+	const siteSelectorRef = useHideSiteSelectorOnFocusOut();
+	const canAccessJetpackManage = useSelector( hasJetpackPartnerAccess );
+
+	return (
+		<BaseSiteSelector
+			forwardRef={ siteSelectorRef }
+			className="jetpack-cloud-sidebar__site-selector"
+			indicator
+			showAddNewSite
+			showAllSites
+			groups
+			/* eslint-disable-next-line jsx-a11y/no-autofocus */
+			autoFocus={ isVisible }
+			isJetpackAgencyDashboard={ canAccessJetpackManage }
```

Comment:
This was `isJetpackAgencyDashboard={ isJetpackManage }` previously. Is this change intentional?

Reply:
This change was intentional on my part, yeah -- `isJetpackAgencyDashboard` controls which "Add new site" button is displayed at the bottom of the selector, and if someone has access to Jetpack Manage we want them to be able to take advantage of its features, regardless of which part of Jetpack Cloud they're in at the moment. (Context: `isJetpackManage` is only set to true on pages that belong to Jetpack Manage itself, which excludes all the single-site pages like Backup, Scan, etc.)